src/subscription_manager/plugins.py

```diff
@@ -157,7 +163,7 @@ class BaseConduit(object):
     slots = []
 
     # clazz is the class object for class instance of the object the hook method maps too
-    def __init__(self, clazz, conf=None):
+    def __init__(self, clazz: type(SubManPlugin), conf: Optional["PluginConfig"] = None):
```

Comment:
Wouldn't be better to use `Type[SubManPlugin]`? Calling `type()` in type hint looks wild.